Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -462,25 +462,47 @@ void BlockFormattingContext::resolve_used_height_if_treated_as_auto(Box const& b
         height = max(height, calculate_inner_height(box, available_space.height, computed_values.min_height()));
     }
 
+    auto writing_mode = box.computed_values().writing_mode();
+    bool has_horizontal_writing_mode = (writing_mode == CSS::WritingMode::HorizontalTb);
+
     if (box.document().in_quirks_mode()
         && box.dom_node()
         && box.dom_node()->is_html_html_element()
-        && box.computed_values().height().is_auto()) {
+        && ((box.computed_values().width().is_auto() && !has_horizontal_writing_mode)
+            || (box.computed_values().height().is_auto() && has_horizontal_writing_mode))) {
```

Comment:
maybe we could simplify this by adding a getter in ComputedValues that returns block/inline size based on writing-mode parameter?